packages/eas-cli/src/user/expoSsoLauncher.ts

```diff
@@ -0,0 +1,82 @@
+import openBrowserAsync from 'better-opn';
+import http from 'http';
+import { Socket } from 'node:net';
+import querystring from 'querystring';
+
+import Log from '../log';
+
+export default (options: {
+  expoWebsiteUrl: string;
+  serverPort: number;
+}): { executeAuthFlow: () => Promise<string> } => {
+  const { expoWebsiteUrl, serverPort } = options;
+  if (!expoWebsiteUrl || !serverPort) {
+    throw new Error('Expo website URL and localserver port are required.');
+  }
+  const scheme = 'http';
+  const hostname = 'localhost';
+  const path = '/auth/callback';
+  const redirectUri = `${scheme}://${hostname}:${serverPort}${path}`;
+
+  const buildExpoSsoLoginUrl = (): string => {
+    const data = {
+      app_redirect_uri: redirectUri,
+    };
+    const params = querystring.stringify(data);
+    return `${expoWebsiteUrl}/sso-login?${params}`;
+  };
+
+  // Start server and begin auth flow
+  const executeAuthFlow = (): Promise<string> => {
+    return new Promise<string>(async (resolve, reject) => {
+      const connections = new Set<Socket>();
+
+      const server = http.createServer(
+        (request: http.IncomingMessage, response: http.ServerResponse) => {
+          try {
+            if (!(request.method === 'GET' && request.url?.includes('/auth/callback'))) {
+              throw new Error('Unexpected SSO login response');
+            }
+            const url = new URL(request.url, `http:${request.headers.host}`);
+            const sessionSecret = url.searchParams.get('session_secret');
+
+            if (!sessionSecret) {
+              throw new Error('Login in the website failed unexpectedly.');
+            }
+            resolve(sessionSecret);
+            response.writeHead(200, { 'Content-Type': 'text/plain' });
+            response.write(`Thank you, website login has completed. You can now close this tab.`);
```

Comment:
is there a way to close the tab programmatically? Either using JS within the response or maybe a method from the library that provides `openBrowserAsync`?

Reply:
I haven't been able to find a way. The library that opens has nothing for close.

I tried responding with `<script>window.close();</script>` both with Content-Type test/plain and text/html, but this just got displayed.

Will look into further later - https://linear.app/expo/issue/ENG-8924/[eas-cli-expo-cli]-make-clis-close-the-browser-tab-after-user